packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts

```diff
@@ -0,0 +1,107 @@
+import { type EthereumWalletConnectorOpts } from '@dynamic-labs/ethereum-core';
+import { EthereumInjectedConnector, type IEthereum } from '@dynamic-labs/ethereum';
+import { toPrivyWalletProvider } from '@privy-io/cross-app-connect'
+import { transformEIP1193Provider } from '@abstract-foundation/agw-client';
+import { abstractTestnet } from 'viem/chains';
+import { DynamicError } from '@dynamic-labs/utils';
+import { type Chain, logger } from '@dynamic-labs/wallet-connector-core';
+import { findWalletBookWallet } from '@dynamic-labs/wallet-book';
+import { toHex } from 'viem';
+const AGW_APP_ID = 'cm04asygd041fmry9zmcyn5o5';
+
+export class AbstractEvmWalletConnector extends EthereumInjectedConnector {
+
+  /**
+   * The name of the wallet connector
+   * @override Required override from the base connector class
+   */
+  override name = 'Abstract';
+
+  /**
+   * The constructor for the connector, with the relevant metadata
+   * @param props The options for the connector
+   */
+  constructor(props: EthereumWalletConnectorOpts) {
+    super({
+      ...props,
+      metadata: {
+        id: 'abstract',
+        name: 'Abstract',
+        icon: 'https://d9s2izusg5pvp.cloudfront.net/icon/light.png',
+      },
+    });
+
+    this.isInitialized = false;
+    this.wallet = findWalletBookWallet(this.walletBook, this.key);
+  }
+  /**
+   * Returns false because we only support Abstract
+   */
+  override supportsNetworkSwitching(): boolean {
+    return false;
+  }
+
+  override isInstalledOnBrowser(): boolean {
+    return true;
+  }
+
+  override async init(): Promise<void> {
```

Comment:
do we need to add a check that abstract chain is enabled? (this come from evmNetworks in the props)